### PR TITLE
CSHARP-3658: Versioned API Connection Examples for Docs

### PR DESCRIPTION
**Pull request by MikalaiMazurenka**
JIRA: https://jira.mongodb.org/browse/CSHARP-3658

### CODE REVIEW

File: tests/MongoDB.Driver.Examples/VersionedApiExamples.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+{
+    public class VersionedApiExamples
+    {
+        public void ConfigureServerApi()
+        {
+            // Start Versioned API Example 1
+            var connectionString = "mongodb://localhost";
+            var serverApi = new ServerApi(ServerApiVersion.V1);
+            var mongoClientSettings = MongoClientSettings.FromConnectionString(connectionString);
+            mongoClientSettings.ServerApi = serverApi; // ServerApi is not available as a connection string option and should be configured on the settings object
```
Comment (jyemin) [inline tests/MongoDB.Driver.Examples/VersionedApiExamples.cs:26]: Let's remove the comment

File: tests/MongoDB.Driver.Examples/VersionedApiExamples.cs [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+            mongoClientSettings.ServerApi = serverApi;
+            var mongoClient = new MongoClient(mongoClientSettings);
+            // End Versioned API Example 2
+        }
+
+        public void ConfigureServerApiNonStrict()
+        {
+            // Start Versioned API Example 3
+            var connectionString = "mongodb://localhost";
+            var serverApi = new ServerApi(ServerApiVersion.V1, strict: false); // Current server default is false, but it can be specified explicitly
```
Comment (jyemin) [inline tests/MongoDB.Driver.Examples/VersionedApiExamples.cs:46]: Let's remove the comment

**Review by jyemin — CHANGES_REQUESTED** [review]

**Review by rstam — APPROVED** [review]
LGTM

I'm fine with Jeff's suggested changes

File: tests/MongoDB.Driver.Examples/VersionedApiExamples.cs (hunk 1 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Examples/VersionedApiExamples.cs:26]: Done.

File: tests/MongoDB.Driver.Examples/VersionedApiExamples.cs (hunk 2 above)
Comment (MikalaiMazurenka) [reply tests/MongoDB.Driver.Examples/VersionedApiExamples.cs:46]: Done.

**Review by MikalaiMazurenka — COMMENTED** [review]
Ready for review!

**Review by jyemin — APPROVED** [review]
LGTM

**Pull request merged**